plugin/src/leiningen/cljs_lambda/zip_tedium.clj

```diff
@@ -2,12 +2,32 @@
   (:require [leiningen.cljs-lambda.logging :refer [log]]
             [clojure.java.io :as io])
   (:import [java.io File]
-           [java.util.zip ZipEntry ZipOutputStream]))
+           [java.nio.file Files LinkOption]
+           [java.nio.file.attribute PosixFilePermission]
+           [org.apache.commons.compress.archivers.zip ZipArchiveEntry
+                                                      ZipArchiveOutputStream]))
+
+(defn- get-posix-mode [file]
```

Comment:
Thanks a whole lot for this - works great.  Do you think this is an improvement?

```clojure
(def ^:private bit->perm
  (map-indexed
   vector
   [PosixFilePermission/OTHERS_EXECUTE
    PosixFilePermission/OTHERS_WRITE
    PosixFilePermission/OTHERS_READ

    PosixFilePermission/GROUP_EXECUTE
    PosixFilePermission/GROUP_WRITE
    PosixFilePermission/GROUP_READ

    PosixFilePermission/OWNER_EXECUTE
    PosixFilePermission/OWNER_WRITE
    PosixFilePermission/OWNER_READ]))

(defn- get-posix-mode [file]
  (let [no-follow (into-array [LinkOption/NOFOLLOW_LINKS])
        perms     (Files/getPosixFilePermissions (.toPath file) no-follow)]
    (reduce
     (fn [mode [bit perm]]
       (cond-> mode
         (contains? perms perm) (bit-set bit)))
     0 bit->perm)))
```